### DISCUSSION TITLE
How to add tooltips to a plot?

### DISCUSSION THREAD

**Discussion by bryevdv** [bokeh/bokeh#12392] [Internals and design (Q&A)]
@mattpap How can a user affix a `Tooltip` to a fixed location on a plot, preferably using data-space coordinates? I tried
```python
from bokeh.models import Tooltip
from bokeh.plotting import figure, curdoc, show

plot = figure()
plot.line([1, 2], [1, 2])

tooltip = Tooltip(
    content="tooltip", visible=True, position=(150, 150), target=plot
)
curdoc().add_root(tooltip)

show(plot)
``` 
expecting to display at screen coordinates, but that did not display anything. I also tried `target=ByCSS("canvas")` as well.

**Comment by mattpap** [comment]
Same problem as in https://github.com/bokeh/bokeh/pull/12370#pullrequestreview-1103321741. `curdoc()` is ignored by embed APIs. Then `show()` doesn't allow sequences, so `save([...])` is the only thing that works:

```py
from bokeh.models import Tooltip
from bokeh.plotting import figure, save

plot = figure(width=300, height=300)
plot.line([1, 2], [1, 2])

tooltip = Tooltip(
    content="tooltip",
    visible=True,
    position=(100, 50),
    target=plot,
)

save([plot, tooltip])
```
![image](https://user-images.githubusercontent.com/27475/190345276-76084c24-b9d2-429f-aa94-2eeb6ec7047c.png)
